src/Mocking.jl

```diff
@@ -209,30 +145,31 @@ macro mock(expr)
     expr.head == :call || error("expression is not a function call")
     ENABLED::Bool || return esc(expr)  # @mock is a no-op when Mocking is not ENABLED
 
-    func = expr.args[1]
-    func_name = QuoteNode(func)
+    target = expr.args[1]
     args = filter(x -> !Mocking.iskwarg(x), expr.args[2:end])
     kwargs = extract_kwargs(expr)
 
-    env_var = gensym("env")
     args_var = gensym("args")
+    alternate_var = gensym("alt")
 
     # Note: The fix to Julia issue #265 (PR #17057) introduced changes where no compiled
     # calls could be made to functions compiled afterwards. Since the `Mocking.apply`
     # do-block syntax compiles the body of the do-block function before evaluating the
     # "outer" function this means our patch functions will be compiled after the "inner"
     # function.
     result = quote
-        local $env_var = Mocking.get_active_env()
         local $args_var = tuple($(args...))
-        if Mocking.ismocked($env_var, $func_name, $args_var)
-            Base.invokelatest($env_var.mod.$func, $args_var...; $(kwargs...))
+        local $alternate_var = Mocking.get_alternate($target, $args_var...)
```

Comment:
do these `local` delcarations do anything?
Would we like to wrapp this in a `let` block?

Reply:
The `local` is mostly to be fully explicit. We could use a `let` block but since we're using `gensym`d names it should be fine.